torch/distributed/elastic/rendezvous/c10d_rendezvous_backend.py

```diff
@@ -166,6 +167,21 @@ def _create_tcp_store(params: RendezvousParameters) -> TCPStore:
 
     return store
 
+def _create_file_store(params: RendezvousParameters) -> FileStore:
+    # If a user specifies an endpoint, we treat it as a path to a file.
+    if params.endpoint:
+        path = params.endpoint
+    else:
+        try:
+            # The temporary file is readable and writable only by the user of
+            # this process.
+            _, path = tempfile.mkstemp()
+        except OSError as exc:
+            raise RendezvousError(
+                "The file creation for C10d store has failed. See inner exception for details."
+            ) from exc
+
+    return FileStore(path)
```

Comment:
Since the endpoint can now be specified by the user as well, we should wrap this call with an exception handler similar to line 179.

Reply:
So right now, it doesn't seem that there is any check on the path that is performed when FileStore is initialized (from looking at FileStore.cpp). It would only fail if an operation such as get/set is attempted to be performed. I'll still wrap this with a runtime/value error, but I wasn't sure.